native/src/main/java/io/ballerina/lib/data/csvdata/utils/CsvUtils.java

```diff
@@ -0,0 +1,373 @@
+        }
+        return key;
+    }
+
+    public static long[] getSkipLinesFromStringConfigValue(String configValue) {
+        String[] parts = configValue.split(SKIP_LINE_RANGE_SEP);
+        if (parts.length != 2) {
+            throw DiagnosticLog.error(DiagnosticErrorCode.INVALID_RANGE_FOR_SKIPLINES);
+        }
+        try {
+            int start = Integer.parseInt(parts[0]);
+            int end = Integer.parseInt(parts[1]);
+            int size = end - start + 1;
+            if (size <= 0) {
+                throw DiagnosticLog.error(DiagnosticErrorCode.INVALID_RANGE_FOR_SKIPLINES);
+            }
+            long[] result = new long[size];
+            for (int i = 0; i < size; i++) {
+                result[i] = start + i;
+            }
+            return result;
+        } catch (NumberFormatException e) {
+            throw DiagnosticLog.error(DiagnosticErrorCode.INVALID_VALUE_FOR_SKIPLINES);
+        }
+    }
+
+    public static long[] getSkipDataRows(Object skipLines) {
+        if (skipLines == null) {
+            return EMPTY_LONG_ARRAY;
+        }
+
+        if (skipLines instanceof BArray skipLinesArray) {
+            if (skipLinesArray.getLength() == 0) {
+                return EMPTY_LONG_ARRAY;
+            }
+            return skipLinesArray.getIntArray();
+        }
+
+        return getSkipLinesFromStringConfigValue(StringUtils.getStringValue(skipLines));
+    }
+
+    public static boolean isNullValue(Object nullValue, Object value) {
+        if (value == null) {
+            return true;
+        }
+        if (value instanceof BString bString) {
+            value = StringUtils.getStringValue(bString);
+        }
+        if (value instanceof String v) {
+            return handleStringNullValue(nullValue, v, value);
+        }
+        return false;
+    }
+
+    private static boolean handleStringNullValue(Object nullValue, String v, Object value) {
+        if ((nullValue == null) && (Constants.Values.NULL.equalsIgnoreCase(v)
+                || Constants.Values.BALLERINA_NULL.equalsIgnoreCase(v))) {
+            return true;
+        }
+        return nullValue != null && value.equals(StringUtils.getStringValue(nullValue));
+    }
+
+    public static boolean isCharContainsInLineTerminatorUserConfig(char c,
+                   Object lineTerminatorObj, boolean isCarriageTokenPresent) {
+        if (lineTerminatorObj instanceof BArray array) {
+            Object[] lineTerminators = array.getValues();
+            for (Object lineTerminator: lineTerminators) {
+                Optional<Boolean> value = handleLineTerminator(lineTerminator, c, isCarriageTokenPresent);
+                if (value.isEmpty()) {
+                    continue;
+                }
+                return value.get();
+            }
+            return false;
+        }
+
+        String lineTerminator = StringUtils.getStringValue(StringUtils.fromString(lineTerminatorObj.toString()));
```

Comment:
It seems to me that this part and `handleLineTerminator`has a lot of duplicated code. I feel like it is better if we have a "inner" function (like `handleLineTerminator`) that handles a single value. Then for array you go over each value in that call this inner function and for the single value case you just call the inner function. WDYT?